### DISCUSSION TITLE
Fallback NodeRenderer

### DISCUSSION THREAD

**Discussion by emad-elsaid** [yuin/goldmark#345] [Ideas]
- I had a situation where I want to render certain node in the AST if it has a specific property
- I would like to render the fenced code block in case the language matches a format such as `alert` or `info`. 
- In case it doesn't match the format that I can handle I want to let the renderer continue rendering using the other lower priority node renderer.

I looked into the part that does the node rendering and noticed it returns an error and stops rendering? 

https://github.com/yuin/goldmark/blob/60df4aadee09cba8a9e28af96d5e9e8883050ff4/renderer/renderer.go#L168

here are my thoughts so far:

- Maybe we can add another walk status `WalkFallback` and check if the node renderer returned it so we can continue to use the next renderer in the list
- Maybe we can have a predefined error that the renderer can use it to also tell us we should continue to render the node.

I'm interested in knowing: 

- how do we feel about this problem? and if you faced it before
- is any of the proposed solutions/changes is viable?

**Comment by yuin** [comment]
I do not stand this idea.

Renderer should stick to rendering nodes. Renderer should not have logics like you suggested.
Have a think about case the following:
If anyone wants to render fenced code blocks whose language is `alert` as a pdf. He must write same logics as you written for HTML renderers.

So, in your case, it will be better that transform AST nodes as  a specific node like`NoticeBlock`. `NoticeBlockRenderer` will just render the nodes without logics.

**Comment by emad-elsaid** [comment]
Good point, I ended up actually doing this in my code here https://github.com/emad-elsaid/xlog/commit/9f31a673c0960da8bdf47e2a518a6af38cc80227
